### PR TITLE
BFD-2134: Add initial ephemeral env support

### PR DESCRIPTION
**Pull request by mjburling**
**JIRA Ticket:**
[BFD-2134](https://jira.cms.gov/browse/BFD-2134)

**User Story or Bug Summary:**

Ensure that existing terraservice modules can accommodate ephemeral environments.

---

### What Does This PR Do?
This PR predominantly focuses on extending the `base` _terraservice_ module to allow users to define non-established environments (e.g. `test`, `prod-sbx`, and `prod`) _or_ more simply, _ephemeral environments_. We took care when introducing the `common` and `migrator` terraservice modules, and as a result, they needed little modification to support an arbitrary environment definitions.

**There is complexity in the `base` module.** However, with the understanding that the objectives for "ephemeral environment support" in the base module are few and humble, this should lend itself to a relatively quick review:
1. copy templated string values from `ephemeral.yaml` into an appropriate, ephemeral environment's AWS SSM Parameter Store hierarchy
2. provide ergonomic, overridable way to resolve the desired RDS cluster snapshot from the "seed" environment
3. copy the necessary "seed" environment configuration, **while** **respecting** **the** **"seed"** **environment's** **security** **posture**

tl;dr; this enables well-credentialed bfd engineers to create arbitrary environments using the `base`, `common`, and `migrator` terraservice modules, which creates a limited _ephemeral environment_ that's suitable for evaluating database migrations **and** general infrastructure improvements.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    6. There isn't too much of a burden on reviewers.
    7. Any problems it causes have a small "blast radius".
    8. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by mackec1 — APPROVED** [review]
Looks good; glad I had the tutorial prior to looking at it!

File: ops/terraform/services/base/values/ephemeral.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# COMMON CONFIGURATION
+/bfd/${env}/common/nonsensitive/enterprise_tools_security_group: bfd-test-enterprise-tools
+/bfd/${env}/common/nonsensitive/env_name_std: ${env}
+/bfd/${env}/common/nonsensitive/key_pair: bfd-${seed_env}
+/bfd/${env}/common/nonsensitive/kms_key_alias: alias/bfd-${seed_env}-cmk
+/bfd/${env}/common/nonsensitive/management_security_group: bfd-test-remote-management
+/bfd/${env}/common/nonsensitive/new_relic_metric_host: &newRelicMetricHost https://gov-metric-api.newrelic.com
+/bfd/${env}/common/nonsensitive/new_relic_metric_path: &newRelicMetricPath /metric/v1
+/bfd/${env}/common/nonsensitive/vpc_name: bfd-test-vpc
+/bfd/${env}/common/nonsensitive/vpn_security_group: bfd-test-vpn-private
```
Comment (christrotteradhoc) [inline ops/terraform/services/base/values/ephemeral.yaml:11]: commenting here about the reasons why we use the test vpc would be nice

**Review by christrotteradhoc — COMMENTED** [review]
looks good, thanks for the walkthrough ; left a request for a comment that might be helpful for others in the future.

**Review by christrotteradhoc — APPROVED** [review]

**Review by mackec1 — APPROVED** [review]
Looks good...thx.
